core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -47,6 +40,12 @@
  */
 public class RoutingNodes implements Iterable<RoutingNode> {
 
+    private final static Random RANDOM;
+
+    static {
+        RANDOM = Randomness.create();
+    }
+
```

Comment:
here is a better suggestion:
```
static Random random;
static {
   if (System.getProperty("tests.seed") != null) {
       random = new Random() {
           // ensure no bits are consumed by e.g. a test class's clinit, since that will never reproduce.
           // randomized runner has a similar check against its random()
           @Override
            protected int next(int bits) {
                throw new IllegalStateException("consuming random bits at illegal time, this will not reproduce!");
            }
       }
   } else {
      random = new Random();
   }
}

/** this is set by ESTestCase.beforeClass, to a new Random instance based on runner seed,
    so each test class is "reset", since tests are run in groups on the same jvm. in ESTestCase.afterClass,
    set set it back to an exception-throwing one again, to fail tests instead of making them non-reproducible */
static void setRandom(Random r) {
  random = r;
}
```
I don't know how many classes you have under this category, but each one would need to be handled in this fashion if you want to repro a single test case that failed in jenkins.